contracts/BaseRouter.sol

```diff
@@ -272,6 +290,21 @@ abstract contract BaseRouter is Initializable {
         }
     }
 
+    /// @notice Wrapper built on top of the base `_deposit4626Referral` function to add a proxy enabling to track
```

Comment:
why do we need this one?
Better to save space for other function

Reply:
Because we don't want to ask wallets to deal with our complex interface and enum when building on top of us